### PR TITLE
cleaned up vectorise

### PR DESCRIPTION
**Pull request by chunkybanana**
The code's a lot cleaner and less repetitive now.

Everything in test_complex.py passes, and it seems to work overall, but I may have broken some other stuff. Review with caution.

### CODE REVIEW

File: vyxal/elements.py [hunk 1]
```diff
@@ -5338,199 +5335,50 @@ def vectorise(
...
-                )
-            ),
-        }
-
-        if explicit:
+    if explicit:
+        # explicit vectorisation
+        # Algorithm: If there are no lists, vectorise over lhs.
+        # Else, vectorise over the first list.
+        ts = [primitive_type(x) for x in args]
```
Comment (Lyxal) [inline vyxal/elements.py:5342]: Rip vectorisation over infinite lists

**Review by Lyxal — APPROVED** [review]

File: vyxal/elements.py [hunk 2]
```diff
@@ -5338,199 +5335,50 @@ def vectorise(
...
-            lhs = iterable(lhs, ctx=ctx)
+        # regular vectorisation
+        @lazylist
+        def vectorise_helper():
+            # Like zip, but repeats scalars
+            @lazylist
+            def row_helper(index):
+                for item in args:
+                    if primitive_type(item) is list:
+                        yield item[index]
```
Comment (Steffan153) [inline vyxal/elements.py:5365]: This fails when doing something like `[1,2,3,4] + [5,6,7]`.

**Review by Steffan153 — CHANGES_REQUESTED** [review]
Whoops forgot to request changes.

**Comment by chunkybanana** [comment]
Well, frick.

File: vyxal/elements.py (hunk 1 above)
Comment (chunkybanana) [reply vyxal/elements.py:5342]: Why? This is iterating over each argument to vectorise. Are you referring to another line?

Comment (Lyxal) [reply vyxal/elements.py:5342]: Oh.


I thought it was going over items in the list that's passed in

**Review by Lyxal — APPROVED** [review]
Gg

**Review by Steffan153 — APPROVED** [review]
lgtm

**Comment by chunkybanana** [comment]
If this fails, blame me.

**Pull request merged**